packages/testkit-backend/src/skipped-tests/skip.js

```diff
@@ -1,21 +1,42 @@
+
+function asComposablePredicate (predicate) {
+  return new Proxy(predicate, {
+    get: (target, p) => {
+      if (p === 'and') {
+        return otherPredicate => asComposablePredicate(testName => target(testName) && otherPredicate(testName))
+      } else if (p === 'or') {
+        return otherPredicate => asComposablePredicate(testName => target(testName) || otherPredicate(testName))
+      }
+      return target[p]
+    }
+  })
+}
+
 export function ifEndsWith (suffix) {
-  return testName => testName.endsWith(suffix)
+  return asComposablePredicate(testName => testName.endsWith(suffix))
 }
 
 export function ifStartsWith (prefix) {
-  return testName => testName.startsWith(prefix)
+  return asComposablePredicate(testName => testName.startsWith(prefix))
 }
 
 export function ifEquals (expectedName) {
-  return testName => testName === expectedName
+  return asComposablePredicate(testName => testName === expectedName)
 }
 
 export function or () {
```

Comment:
Why is there no `and` predicate? Or... is the `or` predicate still needed with `asComposablePredicate`? How does operator precedence work?

Reply:
In the way it is implemented, it will work from the left to right with the same precedence. So, `or` will not shortcut.